tests/misc_env/lvm_deployer.py

```diff
@@ -0,0 +1,86 @@
+from utility.log import Log
+from utility.lvm_utils import pvcreate, vgcreate
+
+LOG = Log(__name__)
+
+
+def find_storage_unit(value):
+    """Return with storage Unit based on the provided value.
+
+    Example::
+
+        if value is 10Gb|10G|10Gi|10g,
+        then return 10G.
+
+        K --> Kilo byte
+        M --> Mega byte
+        G --> Giga byte
+        P --> Peta Byte
+
+    Args:
+        value: storage size with unit.
+    Returns:
+        storage metric value with right unit.
+    """
+    value = value.lower()
+    metric = "".join([i for i in value if i.isdigit()])
+
+    if "k" in value:
+        return f"{metric}K"
+    elif "m" in value:
+        return f"{metric}M"
+    elif "g" in value:
+        return f"{metric}G"
+    elif "t" in value:
+        return f"{metric}T"
+
+
+def run(ceph_cluster, **kw):
+    """Test utility to create LVs based on the configuration.
+
+    Module creates LVs on each node using Custom configuration
+    values and assign newly created LVs to node.volume_list.
+
+    Example::
+
+        python3 run.py .... --custom-config lvm_count=3,lvm_size=100G
+
+    Example::
+
+        - test:
+            name: Deploy LVMs for OSD deployment
+            desc: Deploy LVM using custom configuration
+            module: lvm_deployer.py
+
+    Args:
+        ceph_cluster (ceph.ceph.Ceph): Ceph cluster object
+    """
+    custom_config = kw.get("test_data", {}).get("custom-config")
+    num_of_lvms = 0
+    lvm_size = ""
+
+    for i in custom_config:
+        key, value = i.split("=")
+        if key.lower() == "lvm_count":
+            num_of_lvms = value
+        elif key.lower() == "lvm_size":
+            lvm_size = value
+
+    assert num_of_lvms and lvm_size, """lvm_count or lvm_size not found."""
+
+    for node in ceph_cluster.get_nodes():
+        vg_name = "data_vg"
+        if not hasattr(node, "volume_list"):
+            continue
+        devices = [vol.path for vol in node.volume_list]
+        pvcreate(node, " ".join(devices))
+        vgcreate(node, vg_name, " ".join(devices))
+        lvms = []
+        for i in range(num_of_lvms):
+            node.exec_command(
+                cmd=f"lvcreate -y -L {find_storage_unit(lvm_size)} -n lv{i} {vg_name}",
+                sudo=True,
+            )
+            lvms.append(f"/dev/{vg_name}/lv{i}")
+        node.volume_list = [i for i in set(lvms)]
```

Comment:
Can we opt to do `node.volume_list = deepcopy(lvms)`

Reply:
yes, I was in assumption of multiple node LVMs.

Reply:
done.